api/db/migrate/20220926142122_create_decision_tree.rb

```diff
@@ -0,0 +1,16 @@
+class CreateDecisionTree < ActiveRecord::Migration[7.0]
+  def change
+    create_table :decision_trees do |t|
+      t.references :algorithm, null: false, foreign_key: true
```

Comment:
remplace les t.references par des t.belongs_to

-> https://stackoverflow.com/questions/7788188/what-is-the-difference-between-t-belongs-to-and-t-references-in-rails